internal/gatewayapi/testdata/grpcroute-with-service-match.out.yaml

```diff
@@ -79,16 +79,16 @@ xdsIR:
       - "*"
       isHTTP2: true
       routes:
-      - name: default-grpcroute-1-rule-0-match-0-*
+      - name: default-grpcroute-1-rule-0-match-1-*
```

Comment:
why is the order swapped here ?

Reply:
becasue in here, the routes has been `reversed` before sorting
https://github.com/envoyproxy/gateway/blob/78b62e403b1941b00ecd341e55b0f7e13594e325/internal/gatewayapi/sort.go#L50-L57
and the sorting rule is first matching path, then matching header:
https://github.com/envoyproxy/gateway/blob/78b62e403b1941b00ecd341e55b0f7e13594e325/internal/gatewayapi/sort.go#L18-L30